server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -221,18 +240,23 @@ public int setCurrentVersion(long secretId, long secretContentId, String updater
 
     return dslContext.update(SECRETS)
         .set(SECRETS.CURRENT, secretContentId)
+        .set(SECRETS.EXPIRY, r.getExpiry())
         .set(SECRETS.UPDATEDBY, updater)
         .set(SECRETS.UPDATEDAT, now)
         .where(SECRETS.ID.eq(secretId))
         .execute();
   }
```

Comment:
Because we're explicitly setting the current content version, we always update the series expiration to match.